zellij-server/src/panes/terminal_pane.rs

```diff
@@ -118,63 +162,39 @@ impl Pane for TerminalPane {
-            },
-            [27, 91, 67] => {
-                // right arrow
-                if self.grid.cursor_key_mode {
-                    // please note that in the line below, there is an ANSI escape code (27) at the beginning of the string,
-                    // some editors will not show this
-                    return "OC".as_bytes().to_vec();
-                }
-            },
-            [27, 91, 65] => {
-                // up arrow
-                if self.grid.cursor_key_mode {
-                    // please note that in the line below, there is an ANSI escape code (27) at the beginning of the string,
-                    // some editors will not show this
-                    return "OA".as_bytes().to_vec();
-                }
-            },
+        if self.grid.cursor_key_mode {
+            match input_bytes.as_slice() {
+                LEFT_ARROW => {
+                    return AnsiEncoding::Left.as_vec_bytes();
+                },
+                RIGHT_ARROW => {
+                    return AnsiEncoding::Right.as_vec_bytes();
+                },
+                UP_ARROW => {
+                    return AnsiEncoding::Up.as_vec_bytes();
+                },
+                DOWN_ARROW => {
+                    return AnsiEncoding::Down.as_vec_bytes();
+                },
 
-            [27, 91, 72] => {
-                // home key
-                if self.grid.cursor_key_mode {
-                    return vec![27, 79, 72]; // ESC O H
-                }
-            },
-            [27, 91, 70] => {
-                // end key
-                if self.grid.cursor_key_mode {
-                    return vec![27, 79, 70]; // ESC O F
-                }
-            },
-            [27, 91, 66] => {
-                // down arrow
-                if self.grid.cursor_key_mode {
-                    // please note that in the line below, there is an ANSI escape code (27) at the beginning of the string,
-                    // some editors will not show this
-                    return "OB".as_bytes().to_vec();
-                }
-            },
-            [27, 91, 50, 48, 48, 126] | [27, 91, 50, 48, 49, 126] => {
-                if !self.grid.bracketed_paste_mode {
-                    // Zellij itself operates in bracketed paste mode, so the terminal sends these
-                    // instructions (bracketed paste start and bracketed paste end respectively)
-                    // when pasting input. We only need to make sure not to send them to terminal
-                    // panes who do not work in this mode
-                    return vec![];
-                }
-            },
-            _ => {},
-        };
+                HOME_KEY => {
+                    return AnsiEncoding::Home.as_vec_bytes();
+                },
+                END_KEY => {
+                    return AnsiEncoding::End.as_vec_bytes();
+                },
+                [27, 91, 50, 48, 48, 126] | [27, 91, 50, 48, 49, 126] => {
```

Comment:
Do you plan on refactoring this? If not, what about `BRACKETED_PASTE_BEGIN`
and `BRACKETED_PASTE_END` ?

Reply:
Good point! I didn't knew what those stand for and asked you this when I looked at it.
Should be fixed now.